kbase-extension/static/kbase/js/widgets/appWidgets2/input/selectInput.js

```diff
@@ -163,15 +164,12 @@ define([
         }
 
         function setModelValue(value) {
-            const oldValue = model.value;
-            if (oldValue !== value) {
-                model.value = value;
-            }
+            model.value = value;
             setDisabledValuesFromModel();
-        }
-
-        function resetModelValue() {
-            setModelValue(spec.data.defaultValue);
+            $(ui.getElement('input-container.input')).val(value);
+            if (devMode) {
+                channel.emit('set-value', value);
+            }
```

Comment:
it was impossible to know if this was working or not, so I added in a bus message when the component is in dev mode. The first line ensures that the select2 element has the right value, although it won't update the UI because it hasn't triggered the `change` event; unfortunately triggering the `change` event would set up a cycle with the `handleChange` function.